### PR TITLE
feat(widgets): turn on viewEncapsulation.None

### LINKED ISSUE #2564: CSS Variables: allow styling of ng-Bootstrap by their host angular components css file
### Feature Request description:
Styling ng-Bootstrap widgets can be a challenge. 
Especially with /deep/ ::shadow being deprecated. 
By using CSS variables in the bootstrap styling, consumers of the widgets (i.e. Other angular components, with ViewEncapsulation enabled) could then specify override values for these CSS variables. 

Just think, in the local CSS for your component, you could specify a CSS variable value for the background color of the ngb-accordion, and it would set it 'accordingly' ;0

See Excerpt below:
https://developers.google.com/web/updates/2016/02/css-variables-why-should-you-care?hl=en
"This technique is especially useful for theming Web Components that use Shadow DOM, as custom properties can traverse shadow boundaries. A Web Component author can create an initial design using fallback values, and expose theming “hooks” in the form of custom properties."

### PR DESCRIPTION
**Pull request by fbasso**
The purpose of this PR is to allow a better widgets customization with css.

- Fix #2564 and #1875
- It removes the /deep/ selector used in the datepicker, which is deprecated

### CODE REVIEW

File: src/datepicker/datepicker-navigation-select.scss [hunk 1]
```diff
@@ -1,4 +1,4 @@
-:host>select {
+ngb-datepicker-navigation-select > select {
```
Comment (maxokorokov) [inline src/datepicker/datepicker-navigation-select.scss]: this should be more specific without encapsulation, ex:
```css
ngb-datepicker-navigation-select > .custom-select
```

File: src/tooltip/tooltip.scss [hunk 2]
```diff
@@ -1,32 +1,32 @@
-:host {
-  &.bs-tooltip-top .arrow,
-  &.bs-tooltip-bottom .arrow {
+.bs-tooltip {
```
Comment (maxokorokov) [inline src/tooltip/tooltip.scss]: I guess it should be specific to `ngb-popover-window`, not to any `.bs-tooltip`

File: src/popover/popover.scss [hunk 3]
```diff
@@ -1,35 +1,35 @@
-:host {
-  &.bs-popover-top .arrow,
-  &.bs-popover-bottom .arrow {
+.bs-popover {
```
Comment (maxokorokov) [inline src/popover/popover.scss]: I guess it should be specific to `ngb-popover-window`, not to any `.bs-popover`

**Comment by fbasso** [comment]
Thanks for the review ! The changes have been done.

**Pull request closed without merging**